solana/pyth2wormhole/remote_executor/src/api/execute_posted_vaa.rs

```diff
@@ -0,0 +1,77 @@
+
+use std::str::FromStr;
+
+use borsh::{BorshDeserialize, BorshSerialize};
+use bridge::{VerifySignaturesData, PostVAAData, instructions::verify_signatures, SignatureSet, GuardianSet, instructions::post_vaa, CHAIN_ID_SOLANA};
+use solana_program::{pubkey::Pubkey, program::{invoke, invoke_signed}, instruction::Instruction};
+use solitaire::{FromAccounts, Signer, Mut, Info, Peel, Result as SoliResult, ExecutionContext as SolitaireContext, AccountState, Keyed, Data, Owned, AccountOwner};
+
+use crate::{error::Error, types::ExecutorPayload};
+
+const BRIDGE_PROGRAM_ID : &str = "H3fxXJ86ADW2PNuDDmZJg6mzTtPxkYCpNuQUTgmJ7AjU";
+#[derive(FromAccounts)]
+pub struct ExecuteVaa<'b> {
+    pub payer: Mut<Signer<Info<'b>>>,
+    pub signature_set: Mut<Signer<SignatureSet<'b, { AccountState::MaybeInitialized }>>>,
+    pub guardian_set: GuardianSet<'b, { AccountState::Initialized }>,
+    pub executor_pda : Data<'b, ExecutorAccount, { AccountState::MaybeInitialized}>
+}
+
+pub fn execute_vaa(ctx: &SolitaireContext, accs: &mut ExecuteVaa, data: ExecuteVaaData) -> SoliResult<()> {
+    let bridge_pubkey : Pubkey = Pubkey::from_str(BRIDGE_PROGRAM_ID).unwrap();
+
+    //Verify signatures
+    let verify_signatures_ix = verify_signatures(bridge_pubkey, *accs.payer.key, data.post_vaa_data.guardian_set_index, *accs.signature_set.info().key, data.signatures_data)?; 
```

Comment:
`verify_signatures` is called by the guardian, and shouldn't be called during VAA verification